sdk/src/main/scala/com/horizen/account/receipt/EthereumConsensusDataReceipt.scala

```diff
@@ -188,54 +225,57 @@ object EthereumConsensusDataReceipt{
     result
   }
 
-  def createLogsBloom(receipt : Seq[EvmLog]): Array[Byte] = {
-    // we can create bloom filter out of a log or out of a receipt log list 
-    /* see: https://github.com/ethereum/go-ethereum/blob/55f914a1d764dac4bd37a48173092b1f5c3b186d/core/types/bloom9.go
-
-                // CreateBloom creates a bloom filter out of the give Receipts (+Logs)
-                func CreateBloom(receipts Receipts) Bloom {
-                    buf := make([]byte, 6)
-                    var bin Bloom
-                    for _, receipt := range receipts {
-                        for _, log := range receipt.Logs {
-                            bin.add(log.Address.Bytes(), buf)
-                            for _, b := range log.Topics {
-                                bin.add(b[:], buf)
-                            }
-                        }
-                    }
-                    return bin
-                }
-
-                // add is internal version of Add, which takes a scratch buffer for reuse (needs to be at least 6 bytes)
-                func (b *Bloom) add(d []byte, buf []byte) {
-                    i1, v1, i2, v2, i3, v3 := bloomValues(d, buf)
-                    b[i1] |= v1
-                    b[i2] |= v2
-                    b[i3] |= v3
-                }
-
-                // bloomValues returns the bytes (index-value pairs) to set for the given data
-                func bloomValues(data []byte, hashbuf []byte) (uint, byte, uint, byte, uint, byte) {
-                    sha := hasherPool.Get().(crypto.KeccakState)
-                    sha.Reset()
-                    sha.Write(data)
-                    sha.Read(hashbuf)
-                    hasherPool.Put(sha)
-                    // The actual bits to flip
-                    v1 := byte(1 << (hashbuf[1] & 0x7))
-                    v2 := byte(1 << (hashbuf[3] & 0x7))
-                    v3 := byte(1 << (hashbuf[5] & 0x7))
-                    // The indices for the bytes to OR in
-                    i1 := BloomByteLength - uint((binary.BigEndian.Uint16(hashbuf)&0x7ff)>>3) - 1
-                    i2 := BloomByteLength - uint((binary.BigEndian.Uint16(hashbuf[2:])&0x7ff)>>3) - 1
-                    i3 := BloomByteLength - uint((binary.BigEndian.Uint16(hashbuf[4:])&0x7ff)>>3) - 1
-
-                    return i1, v1, i2, v2, i3, v3
-                }
-
-             */
-    // TODO shall we use libevm implementation?
-    new Array[Byte](256)
+  def bytesToShort(byte1: Byte, byte2: Byte): Short = {
+    ByteBuffer
+      .allocate(2)
+      .order(ByteOrder.BIG_ENDIAN)
+      .put(byte1)
+      .put(byte2)
+      .getShort(0)
+  }
+
+  def getBloomFilterValues(data: Array[Byte]) = {
+    // Copy first 3 pairs of bytes to hashBuffer
+    val hashBuffer = Keccak256.hash(data)
+
+    // Calculate byte values
+    val v1 = 1 << (hashBuffer(1) & 0x7)
+    val v2 = 1 << (hashBuffer(3) & 0x7)
+    val v3 = 1 << (hashBuffer(5) & 0x7)
```

Comment:
For every value here and in other places please setup explicitly the types.
It seems very critical in the code like this. Otherwise, whatever change in the algorithm may lead to the  different type assigned to the value, like int\long\short\etc. And we will not be able to detect it properly on compilation time. So may lead to unpredictable result.